contracts/ColonyTask.sol

```diff
@@ -260,24 +261,23 @@ contract ColonyTask is ColonyStorage, DSMath {
   }
 
   function setTaskDomain(uint256 _id, uint256 _domainId) public
+  self()
   taskExists(_id)
   taskNotFinalized(_id)
   domainExists(_domainId)
   {
-    require(tasks[_id].roles[MANAGER].user == msg.sender);
     tasks[_id].domainId = _domainId;
   }
 
   // TODO: Restrict function visibility to whoever submits the approved Transaction from Client
   // Maybe just the administrator is adequate for the skill?
```

Comment:
You can remove this `todo`. It is solved by introducing the restrictions of `self`